zerver/lib/actions.py

```diff
@@ -2404,6 +2404,13 @@ def fetch_stream_subscriber_user_ids(stream: Stream) -> List[int]:
 
         new_user_ids = [user.id for user in users if (user.id, stream.id) in new_streams]
         subscribed_user_ids = all_subscribers_by_stream[stream.id]
+        new_user_names = [u.full_name for u in users if (u.id, stream.id) in new_streams]
+
+        # send message to stream on subscription of user in private stream
+        if stream.invite_only and subscribed_to_stream(get_system_bot(settings.NOTIFICATION_BOT), stream.id):
+            subcription_message = ", ".join(new_user_names)+" subscribed."  # type: Text
```

Comment:
@timabbott let me know if this message is appropriate or should I update it?

Reply:
It's not ideal.  @rishig can help provide a better version.  We should also be thoughtful about how to tag this for translation properly.

Reply:
This can be sent to the topic "hello", where we currently send system generated messages. (Hopefully that topic name is a constant somewhere; if not, a good first commit would be to refactor that out.)

How about the following?
NAME1 added NAME2 to STREAM_NAME. Welcome!
NAME1 added NAME2 and NAME3 to STREAM_NAME. Welcome!
NAME1 added 3 people to STREAM_NAME. Welcome!

Reply:
Maybe without the "Welcome!"s, actually.